lib/elixir/lib/code/fragment.ex

```diff
@@ -312,7 +318,6 @@ defmodule Code.Fragment do
         if ?@ in extra do
           :none
         else
-          {rest, count} = strip_spaces(rest, count)
```

Comment:
An alternative solution could be to conditionally consume spaces here, like this:

```elixir
# Consume spaces only if the identifier is not finished
{rest, count} =
  case strip_spaces(rest, count) do
    {'.' ++ _ = rest, count} when rest == [] or hd(rest) != ?. -> {rest, count}
    _ -> {rest, count}
  end
```

However, this doesn't seem like the proper place for this.